extensions/azurePublishNew/src/components/shared/wizard/Wizard.tsx

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as React from 'react';
+import styled from '@emotion/styled';
+import { DefaultButton } from 'office-ui-fabric-react';
+export type WizardNavigationState = {
+  nextText?: string;
+  canGoNext?: boolean;
+  showBack?: boolean;
+  backText?: string;
+  canGoBack?: boolean;
+  showCancel?: boolean;
+  cancelText?: string;
+  canCancel?: boolean;
+};
+export type ChangeDirection = 'next' | 'back';
+export type WizardStep = {
+  id: string;
+  title: any;
+  subTitle?: any;
+  navigationState?: WizardNavigationState;
+  onNext?: (step?: WizardStep) => void;
+  onRenderContent: (step: WizardStep) => React.ReactNode;
+  onCancel?: (step?: WizardStep) => void;
+  onBack?: (step?: WizardStep) => void;
+  getNextStepId?: () => string | undefined;
+  getBackStepId?: () => string | undefined;
+};
+
+export type Props = {
+  steps: WizardStep[];
+  firstStepId?: string;
+  navigationState?: WizardNavigationState;
+  onRenderHeader?: (step: WizardStep) => React.ReactNode;
+  onRenderFooter?: (step: WizardStep) => React.ReactNode;
+  getNextStepId?: (currentStep: WizardStep) => string | undefined;
+  getBackStepId?: (currentStep: WizardStep) => string | undefined;
+  onStepChange?: (step: WizardStep, changeDirection: ChangeDirection) => void;
+};
+const Footer = styled.div`
+  flex: auto;
+  flex-grow: 0;
+  background: #ffffff;
+  border-top: 1px solid #edebe9;
+  width: 100%;
+  text-align: right;
+  height: fit-content;
+  padding: 24px 0px 0px;
+`;
+const Content = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  min-height: 430px;
+  height: calc(100vh - 65px);
+`;
+const FooterButton = styled(DefaultButton)`
+  margin: 0 4px;
+`;
+export const Wizard = (props: Props) => {
+  const {
+    steps,
+    firstStepId,
+    navigationState: wizardNavigationState,
+    getNextStepId,
+    onStepChange,
+    onRenderFooter,
+    onRenderHeader,
+  } = props;
+  const [currentStepId, setCurrentStepId] = React.useState<string>(
+    firstStepId || (steps.length > 0 ? steps[0].id : '')
+  );
+
+  const stepIndex = React.useMemo(() => {
+    return steps.findIndex((step) => step.id === currentStepId);
+  }, [steps, currentStepId]);
+
+  const currentStep = React.useMemo(() => {
+    return stepIndex >= 0 && stepIndex < steps.length ? steps[stepIndex] : undefined;
```

Comment:
steps[-1] and step[any index larger than length] returns undefined, no need to do the checks